### PR TITLE
BRITECH-85 Setting up commit message hook

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: githooks/commit-msg [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# regex to validate in commit msg
+commit_regex='(BRITECH-[0-9]+)'
+error_msg="Aborting commit. Your commit message is missing either a JIRA Issue ('BRITECH-1111')"
```
Comment (lclarksl) [inline githooks/commit-msg]: Is there an 'or' to the 'either'?

File: githooks/commit-msg [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# regex to validate in commit msg
+commit_regex='(BRITECH-[0-9]+)'
+error_msg="Aborting commit. Your commit message is missing a JIRA Issue ('BRITECH-1111')"
```
Comment (lclarksl) [inline githooks/commit-msg]: This is being super picky, but should we require a space after the JIRA issue id, for readability?

Comment (thullSL) [inline githooks/commit-msg]: What if the JIRA issue is the last thing write before you take a new line
or put it in (brackets), or you're fixing a consecutive issues: from JIRA-01->JIRA-02
or you are just super excited about JIRA-05!!!?! or it's just the last thing you write

Comment (csscottlogic) [inline githooks/commit-msg]: I'd say we just require "^BRITECH-[0-9]+: ", with a space after the colon, i.e. the commit message begins with the JIRA ticket number

**Comment by csscottlogic** [comment]
Should this be in the speakers repo? It'll be the same hook for all repos, so should probably go in dev-setup?

File: githooks/commit-msg (hunk 2 above)
Comment (thullSL) [inline githooks/commit-msg]: Actually the JIRA integration spec seems to want spaces, not sure how strict this is though
[`<ignored text> <ISSUE_KEY> <ignored text> #<COMMAND> <optional COMMAND_ARGUMENTS>`](https://confluence.atlassian.com/fisheye/using-smart-commits-298976812.html#id-_smart_commits_commands-transition)

**Comment by csscottlogic** [comment]
I'll merge this for now, but we need to have a look at the JIRA docs

**Pull request merged**